From b36642e89f2af62eaf3d1866bdd1ac0637d172ad Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@st.com>
Date: Fri, 29 Jul 2016 17:01:31 +0200
Subject: [PATCH 02/23] v4l2object: force DMABUF by default for capture-io-mode

Enables frame 0-copy in output of decode while using
decodebin or playbin.

Change-Id: Ia1f6d663f4a001e328c852ed1c3340a73deabf7f
Reviewed-on: https://gerrit.st.com/56792
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
---
 sys/v4l2/gstv4l2object.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 549e06f..5de969f 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -2755,6 +2755,13 @@ gst_v4l2_object_setup_pool (GstV4l2Object * v4l2object, GstCaps * caps)
   GST_V4L2_CHECK_OPEN (v4l2object);
   GST_V4L2_CHECK_NOT_ACTIVE (v4l2object);
 
+  if (!V4L2_TYPE_IS_OUTPUT (v4l2object->type) &&
+      v4l2object->req_mode != GST_V4L2_IO_DMABUF) {
+    GST_WARNING_OBJECT (v4l2object->element,
+        "Forcing capture-io-mode to GST_V4L2_IO_DMABUF by default...");
+    v4l2object->req_mode = GST_V4L2_IO_DMABUF;
+  }
+
   /* find transport */
   mode = v4l2object->req_mode;
 
-- 
2.7.4

